tomopy/misc/npmath.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy  as np
+import logging
+import warnings
+
+logger = logging.getLogger(__name__)
+
+
+__author__ = "Chen Zhang"
+__all__    = [ 'gauss',
+               'calc_cummulative_dist',
+               'calc_affine_transform',
+]
+
+
+def gauss(x, *p):
+    """
+    simple Guassian func used for quick curve fit
+
+    Parameters
+    ----------
+    x  :  np.1ndarray
+        1D array for curve fitting
+    p  :  parameter lis t
+        A,          mu,      sigma  =   p
+        magnitude  center     std
+
+    Returns
+    -------
+    1d Gaussian distrobution evaluted at x with p
+    """
+    A, mu, sigma = p
+    return A*np.exp(-(x-mu)**2/(2.*sigma**2))
+
+
+def calc_cummulative_dist(data, steps=None):
+    """
+    Calculated the cumulative distribution from a data array without using
+    binning.  
+    This provides a more robust way to representing the statistics of the 
+    incoming data without worrying about the binning/samping effect.
+
+    Parameters
+    ----------
+    data  :  1D np.array
+        1-D numpy array 
+    steps :  [ None | int ], optional
```

Comment:
How the default value is calculated?

Reply:
The default (steps=None) will use the whole data.  It is close to considering using bin_size=1 or bins=len(data).

Reply:
Ah, ok. Could you add this to the docstring?

Reply:
For the purposes of automatically documenting code (such as [autosummary](http://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html)), please reconsider the content format _in_ the doc strings.  The first line should be a summary *all on one line* with vertical white space for the remaining content.

In this case, change to this

```
+    """
    Calculate the cumulative distribution from a data array without using binning.  

    This provides a more robust way to representing the statistics of the 
    incoming data without worrying about the binning/sampling effect.

    Parameters
...
```

Reply:
I updated the docstring accordingly